### ISSUE TITLE
file has been changed since reading it

### ISSUE THREAD

**Issue by alok** [dojoteef/neomake-autolint#11]
Sometimes this plugin changes the current file to the one in the autolint cache. it then gives this error every time you try to switch files.

**Comment by dojoteef** [comment]
@Alok Can you please provide more information.

* What is the commit hash of the version of plugin you are using (`git rev-parse HEAD`).
* What is your current configuration for Neomake and Neomake Autolint

Could you try to replicate the issue with `let g:neomake_verbose = 3` in your `.vimrc` and only Neomake and Neomake Autolint enabled (no other plugins enabled).

**Comment by alok** [comment]
I'm using the latest versions of both. I'll try to reproduce it.

``` {.vim}
let g:neomake_autolint_sign_column_always = 1
let g:neomake_autolint_cachedir = '/Users/alokbeniwal/.cache'

    autocmd! BufWritePost * Neomake

" Map Ctrl-x and Ctrl-z to navigate the location error list (neomake uses it)

let g:neomake_serialize = 0
let g:neomake_verbose = 0
let g:neomake_list_height = 20
```

On 02.03.17, Nader Akoury wrote:
>@Alok Can you please provide more information.
>
>* What is the commit hash of the version of plugin you are using (`git rev-parse HEAD`).
>* What is your current configuration for Neomake and Neomake Autolint
>
>Could you try to replicate the issue with `let g:neomake_verbose = 3` in your `.vimrc` and only Neomake and Neomake Autolint enabled (no other plugins enabled).
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/dojoteef/neomake-autolint/issues/11#issuecomment-277301087

-- 
-- Alok

**Comment by dojoteef** [comment]
@Alok I'm not sure what the underlying issue is, but I have a suspicion it may be related to [autoload/neomake/autolint/buffer.vim#L43-L54](https://github.com/dojoteef/neomake-autolint/blob/fb08484c2078dabb5310e4c63d3e6f4338c2f42f/autoload/neomake/autolint/buffer.vim#L43-L54). That's my only guess. If you do get it to happen again, can you please run `:ls!` and post the output here. It would confirm or deny my suspicion.

**BTW:** You might want to change to following to make use of event triggers (`:help g:neomake_autolint_events`):

```vim
autocmd! BufWritePost * Neomake
```

**Comment by dojoteef** [comment]
@Alok I was able to replicate the issue. To replicate it you need to open vim from a common ancestor of both the file being edited and the cache directory. This is easily replicated:
1. have your Neomake Autolint cache directory be a sub directory of your home directory (`~/.neomake_cache`)
2. have a file that is in your home directory (`~/myproj/somefile.py`)
3. edit that file from your home directory
```sh
~> vi myproj/somefile.py
```

Please first try to replicate the issue the way I described above and verify that it is what is causing your issue. Then you can pull my fix c7689cc and verify that it fixes the issue for you.

If you are unable to replicate the issue the way I described for the fix does not work for you please let me know and I will reopen the issue. Thanks!
